editoast/src/views/work_schedules.rs

```diff
@@ -175,15 +183,107 @@ async fn create(
     }))
 }
 
+#[derive(Serialize, Deserialize, ToSchema)]
+struct WorkScheduleProjectForm {
+    work_schedule_group_id: i64,
+    #[schema(value_type = Vec<TrackRange>)]
+    path_track_ranges: Vec<CoreTrackRange>,
+}
+
+/// This struct represents the projection of one work_schedule on a path.
+/// it takes the type, start time, end time of the work_schedule
+/// as well as a list of ranges (a, b),
+/// where a and b are the distance of the beginning and end of the track range
+/// relatively to the beginning of the path.
+#[derive(Serialize, Deserialize, ToSchema, PartialEq, Debug)]
+struct WorkScheduleProjection {
+    #[serde(rename = "type")]
+    pub work_schedule_type: WorkScheduleType,
+    pub start_date_time: NaiveDateTime,
+    pub end_date_time: NaiveDateTime,
+    pub path_position_ranges: Vec<(u64, u64)>,
+}
+
+#[utoipa::path(
+    post, path = "",
+    tag = "work_schedules",
+    request_body = WorkScheduleProjectForm,
+    responses(
+        (
+            status = 201,
+            body = Vec<WorkScheduleProjection>,
+            description = "Returns a list of work schedule whose track ranges intersect the given path"
+        ),
+    )
+)]
+async fn project(
+    State(app_state): State<AppState>,
+    Extension(authorizer): AuthorizerExt,
+    Json(WorkScheduleProjectForm {
+        work_schedule_group_id,
+        path_track_ranges,
+    }): Json<WorkScheduleProjectForm>,
+) -> Result<Json<Vec<WorkScheduleProjection>>> {
+    let authorized = authorizer
+        .check_roles([BuiltinRole::WorkScheduleRead].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    // get all work_schedule of the group
+    let db_pool = app_state.db_pool_v2.clone();
+    let conn = &mut db_pool.get().await?;
+    let settings: SelectionSettings<WorkSchedule> = SelectionSettings::new()
+        .filter(move || WorkSchedule::WORK_SCHEDULE_GROUP_ID.eq(work_schedule_group_id));
+    let work_schedules = WorkSchedule::list(conn, settings).await?;
+
+    let projections = work_schedules
+        .into_iter()
+        .filter_map(|ws| {
```

Comment:
This `filter_map` is big, I'd rather have it split into a 
```rust
.map({
  ...;
  path_projection.get_intersections(&path_track_ranges)
})
.filter_map(...)
```

Reply:
done

Reply:
done